### PR TITLE
Add new filer support mongodb

### PR DESCRIPTION
**Pull request by bukton123**
Filer server support MongoDB

### CODE REVIEW

**Comment by chrislusf** [comment]
very nice! Thanks!

Please also adjust scaffold.go?

File: weed/filer2/mongodb/mongodb_store.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+func (store *MongodbStore) ListDirectoryEntries(ctx context.Context, fullpath util.FullPath, startFileName string, inclusive bool, limit int) (entries []*filer2.Entry, err error) {
+
+	var where = bson.M{"directory": string(fullpath), "name": bson.M{"$gt": startFileName,}}
+	if inclusive {
+		where["name"] = bson.M{
+			"$gte": startFileName,
+		}
+	}
+	optLimit := int64(limit)
+	cur, err := store.connect.Database(store.database).Collection(store.collectionName).Find(ctx, where, &options.FindOptions{Limit: &optLimit})
```
Comment (chrislusf) [inline weed/filer2/mongodb/mongodb_store.go]: Do you need something similar to "order by" in SQL?
how to ensure the result is sorted?

Comment (bukton123) [reply weed/filer2/mongodb/mongodb_store.go]: I'm sorry. I forgot to add "order by" to the code. I've already added

**Comment by bukton123** [comment]
I add default config MongoDB to scaffold.go

File: weed/filer2/mongodb/mongodb_store.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+func (store *MongodbStore) connection(uri string, poolSize uint64) (err error) {
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+	opts := options.Client().ApplyURI(uri)
+
+	if poolSize > 0 {
+		opts.SetMaxPoolSize(poolSize)
+	}
+
+	client, err := mongo.Connect(ctx, opts)
+	store.connect = client
```
Comment (chrislusf) [inline weed/filer2/mongodb/mongodb_store.go:59]: how about adding an index here if it does not exist, to optimize for the queries?

Comment (bukton123) [reply weed/filer2/mongodb/mongodb_store.go:59]: OK, I adding create an index

**Pull request merged**